backend/src/models/reports/Tab_14_rpt_P_BudgetSummary.js

```diff
@@ -1,156 +1,195 @@
-      initiation_date: knex.raw(`TO_CHAR(cr.initiation_date :: DATE, '${dateFormatShortYear}')`),
-      summary: "cr.summary",
-      type: knex.raw(`string_agg(crt.crtype_name, ', ')`),
-    })
-    .leftJoin(`${changeRequestTypeLookupTable} as crtl`, { "crtl.change_request_id": "cr.id" })
-    .leftJoin(`${changeRequestTypeTable} as crt`, { "crtl.crtype_id": "crt.id" })
-    .groupBy("cr.id")
-    .where({ "cr.link_id": projectId })
-    .orderBy("cr.version");
-};
+      totalsByFiscal.forEach((fiscalTotal) => {
+        const fiscalYear = fiscalTotal.fiscal_year;
+        const sectionItems = reportSection.filter((section) => section.fiscal_year === fiscalYear);
 
-/**
- * Get the contracts for a specific project by id.
- *
- * @param   {int}   projectId The Project Id for the specific project.
- * @returns {any[]}
- */
-const getContracts = (projectId) => {
-  return knex(`${contractTable} as ct`)
-    .select("*", {
-      supplier: "st.supplier_name",
-      end_date: knex.raw(`TO_CHAR(ct.end_date :: DATE, '${dateFormatShortYear}')`),
-      fiscal: "fy.fiscal_year",
-      contract_amount: knex.raw("ct.total_fee_amount + ct.total_expense_amount"),
-    })
-    .leftJoin(`${supplierTable} as st`, { "st.id": "ct.supplier_id" })
-    .leftJoin(`${fiscalYearTable} as fy`, { "fy.id": "ct.fiscal" })
-    .where({ "ct.project_id": projectId })
-    .orderBy("ct.co_number");
-};
+        withTotals.push({
+          sectionInfo: sectionItems,
+          sectionTotals: fiscalTotal,
+        });
+      });
 
-/**
- * Get the contract totals per fiscal year for a specific project by id.
- *
- * @param   {int}   projectId The Project Id for the specific project.
- * @returns {any[]}
- */
-const getContractSummary = (projectId) => {
-  return knex(`${contractTable} as ct`)
-    .select({
-      fiscal: "fy.fiscal_year",
-      total_contract_amount: knex.raw("SUM(ct.total_fee_amount) + SUM(ct.total_expense_amount)"),
-      total_fee_amount: knex.sum("ct.total_fee_amount"),
-      total_expense_amount: knex.sum("ct.total_expense_amount"),
-    })
-    .leftJoin(`${fiscalYearTable} as fy`, { "fy.id": "ct.fiscal" })
-    .groupBy("fy.fiscal_year")
-    .where({ "ct.project_id": projectId });
-};
+      return withTotals;
+    };
 
-module.exports = {
-  findById,
-  findMostRecentStatusById,
-  getProjectBudget,
-  getDeliverableSummaries,
-  getChangeRequests,
-  getContracts,
-  getContractSummary,
+    return {
+      reportDeliverableWithTotals: combineFiscalTotals(
+        reportDeliverablesTotalsByFiscal,
+        reportDeliverables
+      ),
+      reportDeliverablesGrandTotals,
+      reportProjectInfo,
+      reportStatus,
+      reportCurrentDate,
+      reportChangeRequest,
+      reportContractWithTotals: combineFiscalTotals(
+        reportContractInfoTotalsByFiscal,
+        reportContractInfo
+      ),
+      reportContractInfoGrandTotals,
+    };
+  } catch (error) {
+    log.error(error);
+    throw error;
+  }
 };
+
```

Comment:
now I get why it took so long to finish it lol

Reply:
lol a nightmare